src/rules/selector-no-qualifying-type/index.js

```diff
@@ -0,0 +1,108 @@
+"use strict"
+
+import {
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "selector-no-qualifying-type"
+export const messages = ruleMessages(ruleName, {
+  rejected: "Unexpected selector qualified by type",
+})
+
+export default (enabled, options) => {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: enabled,
+      possible: [ true, false ],
+    }, {
+      actual: options,
+      possible: {
+        ignore: [ "attribute", "class", "id" ],
+      },
+      optional: true,
+    })
+
+    const isElementSelector = (selector) => {
```

Comment:
I can't say that I'm familiar. Will need to look into it. If it does what it sounds like, it may prove useful toward simplifying a number of other plugins :+1: